packages/runtime/test/src/interpreter/opcode-list.ts

```diff
@@ -7563,4 +7566,65 @@ describe("Teal Opcodes", function () {
 			assert.equal(expectedResult, stack.pop());
 		});
 	});
+	describe("Block opcode", function () {
+		const stack = new Stack<StackElem>();
+		let interpreter: Interpreter;
+		const FIRST_VALID = 2000n;
+		this.beforeEach(function () {
+			interpreter = new Interpreter();
+			interpreter.runtime = new Runtime([]);
+			interpreter.runtime.ctx.tx = cloneDeep(TXN_OBJ);
+			interpreter.tealVersion = MaxTEALVersion; // set tealversion to latest (to support all tx fields)
+		});
+
+		it("Should fail when accesing two behind FirstVaild because LastValid is 1000 after", function () {
+			stack.push(BigInt(FIRST_VALID - 2n));
+			const op = new Block(["BlkTimestamp"], 1, interpreter);
+			assert.throws(() => op.execute(stack));
+		});
+
+		it("Should allow to acces two behind FirstValid if LastValid is 100 after", function () {
+			interpreter.runtime.ctx.tx.lv = Number(FIRST_VALID) + 100;
+			stack.push(BigInt(FIRST_VALID - 2n));
+			const op = new Block(["BlkTimestamp"], 1, interpreter);
```

Comment:
You have defined enum for this string. let's use that here.

Reply:
done